lib/transmission-rss/aggregator.rb

```diff
@@ -111,12 +111,15 @@ def process_link(feed, item)
 
       # Link is not a String directly.
       link = link.href if link.class != String
-
+      
+      # Determine whether to use guid or link as seen hash
+      seen_value = feed.seen_by_guid ? (item.guid.content rescue item.guid || link).to_s : link
```

Comment:
This is the main change. The combination of `rescue` and `||` should cover feeds that have guid elements with and without properties (and handle Atom and RSS) as well as fall back to the link if we can't find one at all. There are tests for these cases.